duckduckgo/Android
# Android - change menu actions on pages that were added to Bookmarks or Favorites

Task/Issue URL: https://app.asana.com/0/414730916066338/1200559385909216/f

### Description

This PR adds the ability to remove favorites and edit bookmarks from the overflow menu.
Prior to this, the user could only add favorites and bookmarks from the overflow menu. The actions to "Add Bookmark" /  "add Favorite" were displayed in the overflow menu matter if the visited site was already favorite / bookmarked.

### Steps to test this PR

_Feature `Remove favorite`_

1. Visit a site that was never added as favorite.
2. Tap on the overflow menu and check you see in the list the text "Add Favorite".
3. Tap on "Add favorite". You should see the snackbar at the bottom mentioning the favorite was added.
4. Tap on the overflow menu and check you see in the list the text "Remove Favorite" where "Add Favorite" used to be.
5. Tap on "Remove favorite". You should see the snackbar at the bottom mentioning the site was deleted from favorites.
6. Tap on the overflow menu and check you see in the list the text "Add Favorite".

_Feature `Edit bookmark`_

1. Visit a site that was never bookmarked.
2. Tap on the overflow menu and check you see in the list the text "Add Bookmark".
3. Tap on "Add Bookmark". You should see the snackbar at the bottom mentioning the bookmark was added.
4. Tap on the overflow menu and check you see in the list the text "Edit Bookmark" where "Add Bookmark" used to be.
5. Tap on "Edit Bookmark". You should see a screen that allows you to modify the bookmark details.


_Other tests to make sure the behavior is as expected_

_Removing a favorite / bookmarked site from `Bookmarks` screen_

1. Visit a site that was never added as favorite and was never bookmarked.
2. Tap on the overflow menu and select "Add Favorite"
3. Tap on the overflow menu and select "Add Bookmark"
4. Tap on the overflow menu and check you see "Remove Favorite" and "Edit Bookmark" in the list.
5. From the overflow menu tap on "Bookmarks".
6. Remove the site from both sections: FAVORITES and BOOKMARKS and go back.
7. Tap on the overflow menu and check you see "Add Favorite" and "Add Bookmark" in the list.

_Undo a removed favorite_

1. Visit a site that was previously added as favorite.
2. Tap on the overflow menu and select "Remove Favorite". You should see the snackbar at the bottom mentioning the favorite was deleted.
3. Tap on the "UNDO" action on the snackbar.
4. Tap on the overflow menu and check you still have "Remove Favorite" in the list.


_Import a `bookmarks` file and visit one of the favorite /  bookmarked sites_ 

1. Tap on the overflow menu and select "Bookmarks"
2. On "Bookmarks" screen, tap on the overflow menu and select "Import". Import a bookmarks file and go back.
3. Visit a site that was never added as favorite and was never bookmarked.
4. Tap on the overflow menu and check that you see "Add Bookmark" and "Add Favorite" in the list.
5. Visit a site that was previously favorite / bookmarked (it was present in the file you imported).
6. Tap on the overflow menu and check that you see "Edit Bookmark" and "Remove Favorite" in the list.


### UI changes
Uploaded videos in the ticket as they're too big.
